internal/aws/awsutil/conn.go

```diff
@@ -383,3 +356,51 @@ func getPartition(region string) string {
 	p, _ := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
 	return p.ID()
 }
+
+// order
+// 1. override creds providers
+// 2. shared creds file
+// 3. shared profile
+// explicit keys do not make sense in this context since contrib does not take in explicit keys
+// and cwa does not provide explicit keys to config but come from these methods of getting creds
+// @TODO make this more upstream friendly working in the future
+func customCredentialProvider(cfg *AWSSessionSettings) []credentials.Provider {
+	overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
+	credProviders := make([]credentials.Provider, 0)
+	for _, provider := range overrideCredProviders {
+		for _, file := range cfg.SharedCredentialsFile {
```

Comment:
This is throwing me off. Why are we creating an override credential provider for each of the shared credentials files?

Reply:
Because we pass in odin path through this field. That is how we do it in the current agent. Take a look at how the java agent injects odin.